src/utils/setup/save-asset-files.js

```diff
@@ -9,28 +9,25 @@ const DB = metadata.database;
 
 const CHUNK_SIZE = 500;
 
-const saveFile = async asset => {
-    await fs.mkdir(path.join('static', path.dirname(asset.filename)), {
+const saveFile = async (buffer, filepath) => {
+    await fs.mkdir(path.join('static', path.dirname(filepath)), {
         recursive: true,
     });
-    await fs.writeFile(
-        path.join('static', asset.filename),
-        asset.data.buffer,
-        'binary'
-    );
+    await fs.writeFile(path.join('static', filepath), buffer, 'binary');
 };
 
-// Write all assets to static directory
+// Write all assets to static directory while including all filepaths
 export const saveAssetFiles = async (assets, stitchClient) => {
-    if (assets.length) {
-        const assetQuery = { _id: { $in: assets } };
+    const numAssets = Object.keys(assets).length;
+    if (numAssets) {
+        const assetQuery = { _id: { $in: Array.from(Object.keys(assets)) } };
```

Comment:
[nit] I believe `Object.keys()` [returns an array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#Return_value); is there any reason to create a new instance here?

Reply:
I ran into some issue with it expecting an array and thought it was coming from here. I think you are right, let me go ahead and double-check that